assets/src/edit-story/components/checklist/checklist.js

```diff
@@ -99,51 +105,74 @@ export function Checklist() {
     }
   }, [checkpoint]);
 
+  const visiblePanels = PANEL_VISIBILITY_BY_STATE[checkpoint];
+  const priorityBadgeCount = visiblePanels.includes(ISSUE_TYPES.PRIORITY)
+    ? priorityCount
+    : 0;
+  const designBadgeCount = visiblePanels.includes(ISSUE_TYPES.DESIGN)
+    ? designCount
+    : 0;
+  const accessibilityBadgeCount = visiblePanels.includes(
+    ISSUE_TYPES.ACCESSIBILITY
+  )
+    ? accessibilityCount
+    : 0;
```

Comment:
lifted this state from the `XCheck` components. Now used to calculate the number of visible panels too